packages/react/ds/src/card/card.tsx

```diff
@@ -1,61 +1,136 @@
+  action?: Action;
 };
 
-export const Card = (props: CardProps) => {
-  const { type = 'vertical', title, img, icon, content, actions, href } = props;
-  const layoutClasses = {
-    vertical: 'gi-card gi-card-vertical',
-    horizontal: 'gi-card gi-card-horizontal',
-  };
+// Memoized truncation utility
+const useTruncatedText = (text: string | undefined, maxLength: number) => {
+  return useMemo(() => {
+    if (!text) {
+      return '';
+    }
+    return text.length > maxLength ? `${text.slice(0, maxLength)}...` : text;
+  }, [text, maxLength]);
+};
 
-  return (
-    <div className={layoutClasses[type]}>
-      {img && !icon && (
+const CONTENT_MAX_LENGTH = 120;
+const TITLE_MAX_LENGTH = 60;
+
+export const Card = ({
+  type = 'vertical',
+  title,
+  inset = 'none',
+  subTitle,
+  img,
+  icon,
+  content,
+  action,
+  href,
+  tag,
+}: CardProps) => {
+  const truncatedContent = useTruncatedText(content, CONTENT_MAX_LENGTH);
+  const truncatedTitle = useTruncatedText(title, TITLE_MAX_LENGTH);
+  const truncatedSubTitle = useTruncatedText(subTitle, TITLE_MAX_LENGTH);
+
+  const cardClasses = useMemo(() => {
+    const insetClass = `gi-card-inset-${inset}`;
+    return `gi-card gi-card-${type} ${insetClass}`;
+  }, [type, inset]);
+
+  const renderMedia = () => {
+    if (img && !icon) {
+      return (
         <div className="gi-card-image">
           <a href={href}>
             <img src={img} alt={title} />
           </a>
         </div>
-      )}
-      {icon && (
+      );
+    }
+
+    if (icon) {
+      return (
         <div className="gi-card-icon">
           <a href={href}>
             <Icon {...icon} />
           </a>
         </div>
-      )}
-      <div className="gi-card-content">
-        <div className="gi-card-heading">
-          <Heading as="h5" customClasses="!gi-my-0">
-            {href ? <Link href={href}>{title}</Link> : title}
-          </Heading>
-        </div>
-        <div className="gi-card-paragraph">
-          <Paragraph size="sm">{content}</Paragraph>
+      );
+    }
+
+    return null;
+  };
+
+  const renderTitle = () => {
+    const titleContent = href ? (
+      <Link href={href}>{truncatedTitle}</Link>
+    ) : (
+      truncatedTitle
+    );
+
+    return (
+      <Heading
+        as="h5"
+        customClasses={`!gi-my-0 ${truncatedSubTitle ? '!gi-mb-2' : ''}`}
+      >
+        {titleContent}
+      </Heading>
+    );
+  };
+
+  const renderAction = (action: Action) => {
+    if (action.type === 'link') {
+      return <Link {...action}>{action.children}</Link>;
+    }
+    return <Button {...action}>{action.children}</Button>;
+  };
+
+  return (
+    <div className={cardClasses}>
+      {renderMedia()}
+      <div className={`gi-card-content gi-card-inset-${inset}`}>
+        <div className="gi-card-header">
+          <div className="gi-card-heading">
+            {renderTitle()}
+            {truncatedSubTitle && (
+              <Paragraph size="sm" className="gi-card-subheading">
```

Comment:
Use a `div` not `Paragraph`, otherwise if we change something in the Paragraph the card will be affected.